### ISSUE TITLE
TextInput.on_change does not seem to work

### ISSUE THREAD

**Issue by dima7a14** [beeware/toga#467]
Hello! I just started learning toga and played with tutorial "A slightly less toy example" and came across a problem. Please tell me, it's a bug or I'm doing something wrong.

## Expected Behavior
I added on_change handler to f_input: `f_input.on_change = calculate`. I expect that when you change the value in the field, the function `calculate` will be called.

## Current Behavior
The function `calculate` is not called.

## Your Environment
* Python Version (list the specific version number)
3.5

* Operating System and Version (select from the following and list the specific version number; if your OS is not listed, list that as well)

  - [ ] macOS - version: 
  - [x] Linux - distro: Mint - version: 18.3
  - [ ] Windows - version:
  - [ ] Other - name:  - version:

* Toga Target (the type of app you are trying to generate)
 
  - [ ] android
  - [ ] cocoa
  - [ ] django 
  - [x] gtk
  - [ ] iOS
  - [ ] tvOS
  - [ ] watchOS
  - [ ] winforms 
  - [ ] win32
  - [ ] Other (please specify)

**Comment by westurner** [comment]
The ``TextInput.on_change`` event is not being called when specified at instantiation time either.

Python 3.6, Fedora 28 (gtk)

```python
toga.TextInput(on_change=self.calcuate)
# box.add, main_window.add(box)
```

**Comment by freakboy3742** [comment]
I believe this may be a misreport. `on_change` *does* work - but requires a focus change before it is triggered. If you only have one widget in an app, there's no way to make that focus change happen.

This is definitely something we need to address, if only because it's inconsistent with macOS.

**Comment by DariusMontez** [comment]
It looks like @maranas resolved this issue back in Mar 2019 [here](https://github.com/beeware/toga/commit/68ec17e66e2b82aadffeae9cb1c3ff8cd6c745ef#diff-2d9ed7adde7df0f8477ab430acc1b94e) by subscribing to the `changed` event of the native Gtk widget. From GTK Reference - [the `"changed"` signal](https://developer.gnome.org/gtk3/unstable/GtkEditable.html#GtkEditable-changed)

I installed toga-0.3.0.dev20 this morning and am not experiencing this issue.
My on_change handler is being fired on every keystroke or after a cut or a paste.
Using this app with only a text input: (can be pasted directly into console)

```python
import toga
def build(app):
    return toga.TextInput(on_change=print)

toga.App('app', 'app', startup=build).main_loop()
```

**Comment by freakboy3742** [comment]
I think the credit for the full fix goes to @cachitas in acc206d; but I agree this is now fixed.
